src/registers.rs

```diff
@@ -0,0 +1,89 @@
+const PC_START: u16 = 0x3000;
+
+#[derive(Debug, Clone, Copy, PartialEq)]
+enum RegisterFlags {
+    Pos = 1 << 0,
+    Zro = 1 << 1,
+    Neg = 1 << 2,
+}
+
+const NUM_REGISTERS: usize = 8; // R0-R7
+
+pub struct Registers {
+    regs: [u16; NUM_REGISTERS],
+    pub pc: u16,
+    condition: RegisterFlags,
+}
+
+impl Registers {
+    pub fn new() -> Self {
+        Self {
+            regs: [0; NUM_REGISTERS],
+            pc: PC_START,
+            condition: RegisterFlags::Zro,
+        }
+    }
+
+    /// Get the value of a register without getting clippy warnings
```

Comment:
The part about clippy warnings doesn't add much necessary information.
```suggestion
    /// Get the value of a register
```

Reply:
Trying to farm github badges, huh?